docs/source/examples/Detecting Events.mystnb

```diff
@@ -0,0 +1,386 @@
+
+```{code-cell} ipython3
+gp.update_geos(projection_type="orthographic")
+gp.fig.show()
+```
+
+and voila! The groundtrack terminates almost at the 60 degree latitude mark.
+
+## Eclipse Event
+
+Users can detect umbra/penumbra crossings using the `UmbraEvent` and `PenumbraEvent` event classes,
+respectively. As seen from the above examples, the procedure doesn't change much.
+
+```{code-cell} ipython3
+attractor = Earth
+tof = 2 * u.d
+# Classical orbital elements
+coe = (
+    7000.137 * u.km,
+    0.009 * u.one,
+    87.0 * u.deg,
+    20.0 * u.deg,
+    10.0 * u.deg,
+    0 * u.deg,
+)
+orbit = Orbit.from_classical(attractor, *coe)
+```
+
+Let's search for umbra crossings.
+
+```{code-cell} ipython3
+umbra_event = UmbraEvent(orbit, terminal=True)
+events = [umbra_event]
+
+tofs = np.arange(0, 600, 30) << u.s
+rr, vv = cowell(
+    Earth.k,
+    orbit.r,
+    orbit.v,
+    tofs,
+    events=events,
+)
+print(f"The umbral shadow entry time was {umbra_event.last_t} after the orbit's epoch")
+```
+
+**Note**: Even though the eclipse events `UmbraEvent` and `PenumbraEvent` take the `Orbit` as input, they are not used in propagation but used only to access some helpful attributes of the orbit.
+
+Let us plot the eclipse functions' variation with time.
+
+```{code-cell} ipython3
+k = Earth.k.to_value(u.km ** 3 / u.s ** 2)
+R_sec = Sun.R.to_value(u.km)
+R_pri = Earth.R.to_value(u.km)
+# Position vector of Sun wrt Solar System Barycenter
+r_sec_ssb = get_body_barycentric_posvel("Sun", orbit.epoch)[0]
+r_pri_ssb = get_body_barycentric_posvel("Earth", orbit.epoch)[0]
+
+r_sec = ((r_sec_ssb - r_pri_ssb).xyz << u.km).value
+
+rr = (rr << u.km).value
+vv = (vv << u.km / u.s).value
+
+eclipses = []  # List to store values of eclipse_function.
+for i in range(len(rr)):
+    r = rr[i]
+    v = vv[i]
+    eclipse = eclipse_function(k, np.hstack((r, v)), r_sec, R_sec, R_pri)
+    eclipses.append(eclipse)
+
+plt.xlabel("Time (s)")
+plt.ylabel("Eclipse function")
+plt.title("Eclipse function vs time")
+plt.plot(tofs[:len(rr)].to_value(u.s), eclipses);
+```
+
+<div class="alert alert-info">For simplicity, here we compute the position vector of the primary and the secondary body only once, at the orbit epoch. However, the eclipse events internally recompute the position vectors at each desired instant.</div>
+
+We could get some geometrical insights by plotting the orbit:
+
+```{code-cell} ipython3
+# Plot `Earth` at the instant of event occurence.
+Earth.plot(orbit.epoch+umbra_event.last_t, label="Earth position at event instance");
+
+plotter = StaticOrbitPlotter()
+
+plotter.plot(orbit)
+plotter.set_orbit_frame(orbit)
+
+# Convert satellite coordinates to a `CartesianRepresentation` object.
+coords = CartesianRepresentation(rr[:, 0] << u.km, rr[:, 1] << u.km, rr[:, 2] << u.km)
+plotter.plot_trajectory(coords, color="orange");
+```
+
+It seems our satellite is exiting the umbra region, as is evident from the orange colored trajectory!
+
+## Node Cross Event
```

Comment:
Could you please develop a little bit on this detector? No more than two lines, similarly to other examples you made.

Reply:
Thanks for this! I realized I didn't write much about this event. I added a note about near-equatorial orbits!